MozziGuts.cpp

```diff
@@ -503,6 +555,33 @@ static void startAudioStandard() {
   analogWriteResolution(12);
   analogWrite(AUDIO_CHANNEL_1_PIN, 0);
   tcConfigure(AUDIO_RATE);
+  
+#elif IS_ESP32()
+  #ifdef DO_NOT_INIT_ESP32_I2S
+  #else
+	#if (ESP32_PT8211 == true)
+	i2s_driver_install((i2s_port_t)i2s_num, &i2s_config, 0, NULL);
+	i2s_set_pin((i2s_port_t)i2s_num, &pin_config);
+	//i2s_set_sample_rates((i2s_port_t)i2s_num, 44100);
+	i2s_set_sample_rates((i2s_port_t)i2s_num, AUDIO_RATE);    	
+	i2s_zero_dma_buffer((i2s_port_t)i2s_num);
+	
+	#elif (ESP32_internal == true)
+	i2s_driver_install((i2s_port_t)i2s_num, &i2s_config, 0, NULL); 
+	i2s_set_pin((i2s_port_t)i2s_num, NULL);
+	i2s_set_sample_rates((i2s_port_t)i2s_num, AUDIO_RATE);
+	i2s_zero_dma_buffer((i2s_port_t)i2s_num);
+	i2s_set_dac_mode(I2S_DAC_CHANNEL_RIGHT_EN);
+	#endif
+	
+	timer = timerBegin(0, 80, true);
+	timerAttachInterrupt(timer, &esp32AudioOutput, true);
+		// zc - 30 microseconds is ok for now
+	timerAlarmWrite(timer, 30, true);
```

Comment:
Are you sure, the timer needs to run this fast? On the 8266, I2S already keeps a buffer, I'd be surprised, if it didn't on the ESP32. You can probably write a bunch of samples at once in each pass.

Also, perhaps I can trick you into looking into the I2S output on the 8266? This bypasses the output_buffer, completely, using only the I2S buffer, and I2S output rate, instead. (ESP_AUDIO_OUT_MODE == EXTERNAL_DAC_VIA_I2S)

Reply:
This is the part where it gets real difficult for me. I certainly tried to analyze and replicate the ESP8266 methods but didn't get far with that. For example, i2s_write_lr() doesn't seem to exist for ESP32. So there's a good deal of engineering to be done and it's more than I can handle. The solution I'm using was given by zcarlos in the mozzi google group and I think he pulled it form a web resource elsewhere that I came across too.  

So for the timer interval, I tried different values and nothing gave a better sounding result (mostly garble really) but I don't know in what ballpark range I should be looking. Sound quality with this value (and combination of I2S settings) isn't all that great though.